core/os/thread.h

```diff
@@ -112,24 +112,22 @@ class Thread {
 	static PlatformFunctions platform_functions;
 
 	ID id = UNASSIGNED_ID;
-	static SafeNumeric<uint64_t> id_counter;
-	static thread_local ID caller_id;
+
+	static inline SafeNumeric<uint64_t> id_counter{ 1 };
+	static inline thread_local ID caller_id = id_counter.increment();
 	THREADING_NAMESPACE::thread thread;
 
 	static void callback(ID p_caller_id, const Settings &p_settings, Thread::Callback p_callback, void *p_userdata);
 
 	static void make_main_thread() { caller_id = MAIN_ID; }
-	static void release_main_thread() { caller_id = UNASSIGNED_ID; }
+	static void release_main_thread() { caller_id = id_counter.increment(); }
```

Comment:
Just spotted this. I guess we can leave this line untouched for now.

Reply:
I'm not sure I understand?

Reply:
Oh, nevermind. I hadn't realized the real purpose of this. I was thinking, "Well, we can keep using `UNASSIGNED_ID` for the no longer main thread." However, I now realize assigning a regular id is better (still "unmarks" the thread as main, plus it also gives some unique id).

TL;DR Nevermind. 😅

Reply:
Yeah, without this we'd have to reintroduce the check in the hot path! Which was the point! :laughing: